liquigraph-spring-boot-starter/src/test/java/org/liquigraph/spring/LiquigraphAutoconfigurationTest.java

```diff
@@ -0,0 +1,201 @@
+                .isNotEmpty();
+            assertThatMigrationsHaveRun();
+        }
+    }
+
+    @Test
+    public void does_not_inject_bean_when_disabled() {
+        try (AnnotationConfigApplicationContext context = new AnnotationConfigApplicationContext()) {
+            loadContext(context, "liquigraph.enabled=false");
+
+            assertThat(context.getBeansOfType(SpringLiquigraph.class))
+                .as("SpringLiquigraph is not injected when Liquigraph is disabled")
+                .isEmpty();
+            assertThatMigrationsHaveNotRun();
+        }
+    }
+
+    @Test
+    public void executes_only_configured_execution_contexts() {
+        try (AnnotationConfigApplicationContext context = new AnnotationConfigApplicationContext()) {
+            loadContext(
+                context,
+                "liquigraph.changelog=classpath:/db/liquigraph/changelog-with-contexts.xml",
+                "liquigraph.executionContexts=foo,bar",
+                "liquigraph.url=" + jdbcUrl());
+
+            assertThatMigrationsHaveRun();
+        }
+    }
+
+    private static void loadContext(AnnotationConfigApplicationContext baseContext, Class<?> configuration, String... properties) {
+        setUpEnvironment(baseContext, properties);
+        baseContext.register(configuration);
+        baseContext.refresh();
+    }
+
+    private static void loadContext(AnnotationConfigApplicationContext baseContext, String... properties) {
+        setUpEnvironment(baseContext, properties);
+        baseContext.refresh();
+    }
+
+    private static void setUpEnvironment(AnnotationConfigApplicationContext baseContext, String... properties) {
+        for (String property : properties) {
+            EnvironmentTestUtils.addEnvironment(baseContext, property);
+        }
+        baseContext.register(LiquigraphAutoConfiguration.class);
+    }
+
+    private static void assertThatMigrationsHaveRun() {
+        try (Result result = graphDb().execute("MATCH (n:Sentence {text:'Hello world!'}) RETURN COUNT(n) AS count")) {
+            assertThat(result.hasNext()).as("Query returns a count").isTrue();
+            assertThat(result.next().get("count")).as("There is only 1 sentence in the graph").isEqualTo(1L);
+            assertThat(result.hasNext()).as("No more counts are returned").isFalse();
+        }
+    }
+
+    private static void assertThatMigrationsHaveNotRun() {
+        try (Result result = graphDb().execute("MATCH (n) RETURN COUNT(n) AS count")) {
+            assertThat(result.hasNext()).as("Query returns a count").isTrue();
+            assertThat(result.next().get("count")).as("The graph is empty").isEqualTo(0L);
+            assertThat(result.hasNext()).as("No more counts are returned").isFalse();
+        }
+    }
+
+    private static GraphDatabaseService graphDb() {
+        return neo4j.getGraphDatabaseService();
+    }
+
+    private static String jdbcUrl() {
+        return "jdbc:neo4j:" + neo4j.httpURI().toString();
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "jdbc:neo4j:" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)